lxd/archive/archive.go

```diff
@@ -0,0 +1,71 @@
+package archive
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/lxc/lxd/lxd/apparmor"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+func ExtractWithFds(name string, args []string, stdin io.Reader, stdout io.Writer, s *state.State, profileName string) error {
+	apparmor.ArchiveLoad(s, profileName)
+	defer apparmor.ArchiveUnload(s, profileName)
+
+	p, err := subprocess.NewProcessWithFds(name, args, stdin, stdout)
+	if err != nil {
+		return fmt.Errorf("Failed to start unpack: Failed to creating subprocess: %w", err)
+	}
+
+	p.SetApparmor(apparmor.ArchiveProfileName(profileName))
```

Comment:
Should we only call this if the `apparmor.ArchiveLoad` succeeded? Also, if it did succeed and this call fails, should we be returning an error rather than continuing?

Reply:
This is called only when `ArchiveLoad` succeeded right now. `SetApparmor` doesn't return an error.